modules/objc/VectorSeachSnippets.m

```diff
@@ -0,0 +1,143 @@
+//
+//  VectorSeachSnippets.m
+//
+//  Created by Vlad Velicu on 20/03/2024.
+//
+
+#import <Foundation/Foundation.h>
+#import <CouchbaseLite/CouchbaseLite.h>
+#import <NaturalLanguage/NLEmbedding.h>
+
+CBLDatabase* database;
+CBLCollection* collection;
+NLEmbedding* model;
+
+@interface VectorSearchSnippets : NSObject
+
+@end
+
+@implementation VectorSearchSnippets
+
+// MARK: Configuring a project to use Vector Search.
+
+/*/
+ Download Couchbase Lite iOS Obj-C and Couchbase Lite Vector Search libraries
+ Extract the xcframeworks and copy them into the App Project Directory
+ 
+ Add both libraries to the *Frameworks, Libraries and Embedded Content* of your desired target
+ */
+
+- (void) createDefaultIndex {
+    NSError* error;
+    database = [[CBLDatabase alloc] initWithName:@"vectorDB" error:&error];
+    collection = [database defaultCollection: &error];
+    
+    // MARK: Create a default Vector Index Configuration
+
+    CBLVectorIndexConfiguration* config = [[CBLVectorIndexConfiguration alloc] initWithExpression: @"vector" dimensions: 300 centroids: 20];
+    
+    // Default values already set:
+    NSLog(@"%@", config.encoding); // CBLScalarQuantizer: SQ8
+    NSLog(@"%u", config.metric); // 0 = euclidean
+    NSLog(@"%u", config.minTrainingSize); // 25 * centroids(20)
+    NSLog(@"%u", config.maxTrainingSize); // 256 * centroids(20)
+    
+    // MARK: Set custom settings
+    config.encoding = [CBLVectorEncoding none];
+    config.metric = kCBLDistanceMetricCosine;
+    config.minTrainingSize = 50;
+    config.maxTrainingSize = 300;
+}
+
+- (CBLQueryResultSet*) vectorIndexEmbedding {
+    NSError* error;
+    
+    // MARK: Create Vector Index with Embedding
+    CBLVectorIndexConfiguration* config = [[CBLVectorIndexConfiguration alloc] initWithExpression: @"vector" dimensions: 300 centroids: 20];
+    [collection createIndexWithName: @"vector_index" config: config error: &error];
+    
+    model = [NLEmbedding wordEmbeddingForLanguage: @"english"];
```

Comment:
The code from this line is not relavent. There is already `useVectorMatch()` sameple. Some if these here can be moved there.